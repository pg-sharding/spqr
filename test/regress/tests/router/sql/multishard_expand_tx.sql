\c spqr-console

-- check that numeric type works
CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
CREATE KEY RANGE FROM 300 ROUTE TO sh4 FOR DISTRIBUTION ds1;
CREATE KEY RANGE FROM 200 ROUTE TO sh3 FOR DISTRIBUTION ds1;
CREATE KEY RANGE FROM 100 ROUTE TO sh2 FOR DISTRIBUTION ds1;
CREATE KEY RANGE FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;

ALTER DISTRIBUTION ds1 ATTACH RELATION xxm_expd DISTRIBUTION KEY id;

\c regress

CREATE TABLE xxm_expd(id int, j INT);

COPY xxm_expd (id, j) FROM STDIN;
1	1
2	2
3	3
4	4
5	5
50	50
51	51
100	100
101	101
199	199
200	200
201	201
299	299
300	300
301	301
399	399
\.


BEGIN;
UPDATE xxm_expd SET j = j + 1 WHERE id = 1 /* __spqr__engine_v2: true */;
UPDATE xxm_expd SET j = j + 1 WHERE id = 101 /* __spqr__engine_v2: true */;
UPDATE xxm_expd SET j = j + 1 WHERE id = 299 /* __spqr__engine_v2: true */;
UPDATE xxm_expd SET j = j + 1 WHERE id = 399 /* __spqr__engine_v2: true */;
COMMIT;

BEGIN;
INSERT INTO xxm_expd (id, j) VALUES(55, 55) /* __spqr__engine_v2: true */;
INSERT INTO xxm_expd (id, j) VALUES(155, 155) /* __spqr__engine_v2: true */;
INSERT INTO xxm_expd (id, j) VALUES(355, 355) /* __spqr__engine_v2: true */;
ROLLBACK;

BEGIN;
DELETE FROM xxm_expd /* __spqr__engine_v2: true */;
SELECT * FROM xxm_expd WHERE id = 12 /* __spqr__engine_v2: true */;
SELECT * FROM xxm_expd WHERE id = 212 /* __spqr__engine_v2: true */;
ROLLBACK;

BEGIN;
SELECT * FROM xxm_expd WHERE id = 12 /* __spqr__engine_v2: true */;
SELECT * FROM xxm_expd WHERE id = 212 /* __spqr__engine_v2: true */;
DELETE FROM xxm_expd /* __spqr__engine_v2: true */;
ROLLBACK;

SELECT * FROM xxm_expd ORDER BY id;

UPDATE xxm_expd SET id = -1 /* __spqr__engine_v2: true */;
DELETE FROM xxm_expd /* __spqr__engine_v2: true */;

DROP TABLE xxm_expd;

\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
