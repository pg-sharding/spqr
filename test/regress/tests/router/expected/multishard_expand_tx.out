\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

-- check that numeric type works
CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE FROM 300 ROUTE TO sh4 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 300
(1 row)

CREATE KEY RANGE FROM 200 ROUTE TO sh3 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 200
(1 row)

CREATE KEY RANGE FROM 100 ROUTE TO sh2 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 100
(1 row)

CREATE KEY RANGE FROM 0 ROUTE TO sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 0
(1 row)

ALTER DISTRIBUTION ds1 ATTACH RELATION xxm_expd DISTRIBUTION KEY id;
        attach table         
-----------------------------
 relation name   -> xxm_expd
 distribution id -> ds1
(2 rows)

\c regress
CREATE TABLE xxm_expd(id int, j INT);
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
COPY xxm_expd (id, j) FROM STDIN;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
BEGIN;
UPDATE xxm_expd SET j = j + 1 WHERE id = 1 /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1
UPDATE xxm_expd SET j = j + 1 WHERE id = 101 /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh2
UPDATE xxm_expd SET j = j + 1 WHERE id = 299 /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh3
UPDATE xxm_expd SET j = j + 1 WHERE id = 399 /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh4
COMMIT;
BEGIN;
INSERT INTO xxm_expd (id, j) VALUES(55, 55) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1
INSERT INTO xxm_expd (id, j) VALUES(155, 155) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh2
INSERT INTO xxm_expd (id, j) VALUES(355, 355) /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh4
ROLLBACK;
BEGIN;
DELETE FROM xxm_expd /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
SELECT * FROM xxm_expd WHERE id = 12 /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1
 id | j 
----+---
(0 rows)

SELECT * FROM xxm_expd WHERE id = 212 /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh3
 id | j 
----+---
(0 rows)

ROLLBACK;
BEGIN;
SELECT * FROM xxm_expd WHERE id = 12 /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1
 id | j 
----+---
(0 rows)

SELECT * FROM xxm_expd WHERE id = 212 /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh3
 id | j 
----+---
(0 rows)

DELETE FROM xxm_expd /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
ROLLBACK;
SELECT * FROM xxm_expd ORDER BY id;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
 id  |  j  
-----+-----
   1 |   2
   2 |   2
   3 |   3
   4 |   4
   5 |   5
  50 |  50
  51 |  51
 100 | 100
 101 | 102
 199 | 199
 200 | 200
 201 | 201
 299 | 300
 300 | 300
 301 | 301
 399 | 400
(16 rows)

UPDATE xxm_expd SET id = -1 /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
DELETE FROM xxm_expd /* __spqr__engine_v2: true */;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
DROP TABLE xxm_expd;
NOTICE: send query to shard(s) : sh1,sh2,sh3,sh4
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

